05/DB.test.js

```diff
@@ -1 +1,133 @@
-import DB from './DB';
\ No newline at end of file
+import DB from './DB';
+
+let DBClass;
+
+beforeEach(() => {
+	DBClass = new DB();
+});
+describe('DB insert func', () => {
+	it('func insert return correctly insert data', async () => {
+		const objectTest = {
+			say: 'hello',
+			id: 10,
+		};
+		await DBClass.insert(objectTest);
+
+		const rows = await DBClass.getRows();
+		const containsObjectTest = rows.some(row => row.id === objectTest.id);
+
+		expect(containsObjectTest).toBe(true);
+	});
+
+	it('func insert should throw an error when id is a string', async () => {
+		const objectTest = {
+			say: 'hello',
+			id: 'id',
+		};
+
+		try {
+			await DBClass.insert(objectTest);
+		} catch (error) {
+			expect(error).toBe('ID can be only number!');
+		}
+	});
+
+	it('func insert should throw an error when id is duplicated', async () => {
+		const objectTest = {
+			say: 'hello',
+			id: 2,
+		};
+
+		try {
+			await DBClass.insert(objectTest);
+		} catch (error) {
+			expect(error).toBe("ID can't be duplicated!");
+		}
+	});
+});
+
+describe('DB func select', () => {
+	it('func select should return searching id', async () => {
+		//2 added in DB.js file
+		const id = 2;
+		const result = await DBClass.select(id);
+
+		expect(result.id).toBe(id);
+	});
+
+	it('func select should return reject when id is not finded', async () => {
+		const id = 3;
+
+		try {
+			await DBClass.select(id);
+		} catch (error) {
+			expect(error).toBe('ID not found');
+		}
+	});
+});
+
+describe('DB func remove', () => {
+	it('func remove is working correctly', async () => {
+		//2 added in DB.js file
+		const id = 2;
+
+		const remove = await DBClass.remove(id);
+
+		expect(remove).toBe('Item was remove!');
```

Comment:
Warto też sprawdzić czy faktycznie tego elementu nie ma w bazie.
Sam komunikat nie musi oznaczać, że element został usunięty.
Pewność będziemy mieli jeśli go tam nie będzie.

PS. Modyfikacja w docelowej klasie jest niedopuszczalna. Teraz każdy korzystający z tej klasy będzie miał 1 element w bazie przy inicjalizacji. Jeśli go usunie to testy nie będą mu przechodzić. Takie podejście (dodanie elementu) jest bardzo złe.